whois-api/src/main/java/net/ripe/db/whois/api/rest/search/RpkiRoaMessageGenerator.java

```diff
@@ -0,0 +1,66 @@
+package net.ripe.db.whois.api.rest.search;
+
+import com.google.common.collect.ImmutableList;
+import net.ripe.db.whois.api.rest.domain.Flag;
+import net.ripe.db.whois.api.rest.domain.Flags;
+import net.ripe.db.whois.api.rest.domain.ObjectMessage;
+import net.ripe.db.whois.api.rest.domain.Parameters;
+import net.ripe.db.whois.common.rpki.Roa;
+import net.ripe.db.whois.common.rpki.RpkiDataProvider;
+import net.ripe.db.whois.common.rpki.RpkiService;
+import net.ripe.db.whois.common.rpki.WhoisRoaChecker;
+import net.ripe.db.whois.common.rpsl.ObjectType;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import net.ripe.db.whois.query.QueryMessages;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import static net.ripe.db.whois.query.QueryFlag.ROA_VALIDATION;
+
+@Component
+public class RpkiRoaMessageGenerator implements QueryMessageGenerator {
+
+    private static final ImmutableList<ObjectType> TYPES = ImmutableList.of(ObjectType.ROUTE, ObjectType.ROUTE6);
+    private final boolean isEnabled;
+
+    private final RpkiDataProvider rpkiDataProvider;
+
+    @Autowired
+    public RpkiRoaMessageGenerator(@Value("${roa.validator.available:false}") boolean isRoaChecker, final RpkiDataProvider dataProvider) {
+        this.isEnabled = isRoaChecker;
+        this.rpkiDataProvider = dataProvider;
+    }
+
+    @Override
+    public ObjectMessage proceed(RpslObject rpslObject, Parameters parameters) {
+        if (!canProceed(parameters)){
+            return null;
+        }
+        return validateRoa(rpslObject);
+    }
+
+    @Override
+    public ImmutableList<ObjectType> getTypes() {
+        return TYPES;
+    }
+
+    private boolean canProceed(final Parameters parameters){
+        return isEnabled && hasRoaValidationFlag(parameters.getFlags());
+    }
+
+    private boolean hasRoaValidationFlag(final Flags flags){
+        return flags != null && flags.getFlags().contains(new Flag(ROA_VALIDATION));
+    }
+
+    private ObjectMessage validateRoa(final RpslObject rpslObject){
+        final Roa rpkiRoa = new WhoisRoaChecker(new RpkiService(rpkiDataProvider)).validateAndGetInvalidRoa(rpslObject);
```

Comment:
oh yes sorry I marked as @Service and forgot to change this

Reply:
I will remove @Service from RpkiService...because we are using different RpkiDataProvider (one for testing and another for deploy)...and it was like this before in whois-internal (without the @Service)

Reply:
and I would need to refactor whois-internal if I change that as well...because as I said before we were not using it as a Spring component

Reply:
in that case may be revert to original , whichever is easier for you

Reply:
already done in last update. I can create a task in the backlog to do the refactor. I will do that

Reply:
Task in the backlog [DB-5349](https://jira.ripe.net/browse/DB-5349)

Reply:
I did transform that into a @Component and the service into a @Service 
because of performance issues